Feature/Cash Flow statement (#69)
fivetran/dbt_quickbooks: README.md

| [quickbooks__profit_and_loss](https://github.com/fivetran/dbt_quickbooks/blob/master/models/quickbooks__profit_and_loss.sql) | Table containing all revenue and expense account classes by calendar year and month enriched with account type, class, and parent information, as well as ordering configuration--[scroll below for details](https://github.com/fivetran/dbt_quickbooks/blob/main/README.md#customize-the-account-ordering-of-your-financial-statement-models). |
| [quickbooks__balance_sheet](https://github.com/fivetran/dbt_quickbooks/blob/master/models/quickbooks__balance_sheet.sql) | Table containing all asset, liability, and equity account classes by calendar year and month enriched with account type, class, and parent information, as well as ordering configuration--[scroll below for details](https://github.com/fivetran/dbt_quickbooks/blob/main/README.md#customize-the-account-ordering-of-your-financial-statement-models). |

> | [quickbooks__cash_flow_statement](https://github.com/fivetran/dbt_quickbooks/blob/master/models/quickbooks__cash_flow_statement.sql) | Table containing all cash or cash equivalents, investing, operating, and financing cash flow types by calendar year and month enriched with account type, class, and parent information, as well as ordering configuration. IMPORTANT: It is very likely you will need to configure the cash flow types for your own unique use case. [Scroll below to get full instructions for how to configure your cash flow types](https://github.com/fivetran/dbt_quickbooks/blob/main/README.md#customize-the-cash-flow-types-and-account-ordering-of-your-cash-flow-statement). |

Reviewer: ```suggestion
| [quickbooks__cash_flow_statement](https://github.com/fivetran/dbt_quickbooks/blob/master/models/quickbooks__cash_flow_statement.sql) | Table containing all cash or cash equivalents, investing, operating, and financing cash flow types by calendar year and month enriched with account type, class, and parent information, as well as ordering configuration. **IMPORTANT**: It is very likely you will need to configure the cash flow types for your own unique use case. [Scroll below to get full instructions for how to configure your cash flow types](https://github.com/fivetran/dbt_quickbooks/blob/main/README.md#customize-the-cash-flow-types-and-account-ordering-of-your-cash-flow-statement). |
```

Reviewer: In addition to my suggestion above, I would highlight in here that we are by default using the Indirect Method for generating the cash flow statement.

    quickbooks_union_databases: ['quickbooks_usa','quickbooks_canada'] # use this if the data is in different databases/projects but uses the same schema name
```

### Customize the cash flow types and account ordering of your cash flow statement

> IMPORTANT: It is very likely you will need to reconfigure your `cash_flow_type` to make sure your cash flow statement matches your specific use case. Please examine the following instructions.

Reviewer: ```suggestion
**IMPORTANT**: It is very likely you will need to reconfigure your `cash_flow_type` to make sure your cash flow statement matches your specific use case. Please examine the following instructions.
```

> The current default numbering for ordinals and default cash flow types are set in [the `int_quickbooks__cash_flow_classifications`](https://github.com/fivetran/dbt_quickbooks/blob/main/models/intermediate/int_quickbooks__cash_flow_classifications.sql) model. It' based on best practices for cash flow statements in accounting. You can see these ordinals being implemented in the `quickbooks__cash_flow_classifications` model, then implemented in the `quickbooks__cash_flow_statement` model. The `cash_flow_type` value is assigned off of `account_class`, `account_name` or `account_type`, and the cash flow ordinal is assigned off of `cash_flow_type`.

Reviewer: ```suggestion
The current default numbering for ordinals and default cash flow types are set in [the `int_quickbooks__cash_flow_classifications`](https://github.com/fivetran/dbt_quickbooks/blob/main/models/intermediate/int_quickbooks__cash_flow_classifications.sql) model. It's based on best practices for cash flow statements leveraging the indirect method in accounting. You can see these ordinals being created in the `int_quickbooks__cash_flow_classifications` model, then implemented in the `quickbooks__cash_flow_statement` model. The `cash_flow_type` value is assigned off of `account_class`, `account_name` or `account_type`, and the cash flow ordinal is assigned off of `cash_flow_type`.
```

If you'd like to modify either of these configurations, take the following steps to configure the fields you'd like to modify:

> 1) Import a csv with fields into root (not the dbt package) `seeds` folder, then configure your `cash_flow_statement_type_ordinal` variable in your `dbt_project.yml` to reference the seed file name. 

Reviewer: ```suggestion
1) Create a csv file within your root (not the dbt package) `seeds` folder, then configure your `cash_flow_statement_type_ordinal` variable in your `dbt_project.yml` to reference the seed file name. 
```

- For example, if you created a seed file named `quickbooks_cash_flow_types_ordinals.csv`, then you would edit the `cash_flow_statement_type_ordinal` in your root `dbt_project.yml` as such.
 
  ```yml
  vars:

>      cash_flow_statement_type_ordinal: "{ ref('quickbooks_cash_flow_types_ordinals') }"

Reviewer: ```suggestion
     cash_flow_statement_type_ordinal: "{{ ref('quickbooks_cash_flow_types_ordinals') }}"
```

Reviewer: Can you also update this for the `financial_statement_ordinal` instructions below.

- REQUIRED: Every row should have a non-null `ordinal` and `cash_flow_type` column value. 
- REQUIRED: In each row of the seed file, only populate **ONE** of the `account_class`, `account_type`, `account_sub_type`, and `account_number` columns to avoid duplicated ordinals and cash flow types and test failures. This should also make the logic cleaner in defining which account value takes precedence in the ordering hierarchy. 
- In `cash_flow_statement_type_ordinal_example`, we recommend creating ordinals for each `cash_flow_type` value available (the default types are `Cash or Cash Equivalents`, `Operating`, `Investing`, `Financing`, but you can configure as you like in your seed file) to make sure each cash flow statement line has an ordinal assigned to it. Then you can create any additional customization as needed with the more specific account fields to order even further.   
- In `cash_flow_statement_type_ordinal_example`, the `report` field should always be `Cash Flow`.  

> 

Reviewer: Can we also add a piece here that encourages users to share their experiences within the Fivetran community user group.

Author: Done, adding it to the `financial_statement_ordinal` section as well!
